src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/RowDefinitionCollectionConverter.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Windows.Controls;
+using System.Windows.Markup;
+using System.Globalization;
+using System.Text;
+using MS.Internal;
+
+namespace System.Windows.Controls
+{
+    internal sealed class RowDefinitionCollectionConverter : TypeConverter
+    {
+        #region Public Methods
+
+        /// <summary>
+        /// CanConvertFrom - Returns whether or not this class can convert from a given type.
+        /// </summary>
+        /// <returns>
+        /// bool - True if this converter can convert from the provided type, false if not.
+        /// </returns>
+        /// <param name="typeDescriptorContext"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="sourceType"> The Type being queried for support. </param>
+        public override bool CanConvertFrom(ITypeDescriptorContext typeDescriptorContext, Type sourceType)
+        {
+            return sourceType == typeof(string);
+        }
+
+        /// <summary>
+        /// CanConvertTo - Returns whether or not this class can convert to a given type.
+        /// </summary>
+        /// <returns>
+        /// bool - True if this converter can convert to the provided type, false if not.
+        /// </returns>
+        /// <param name="typeDescriptorContext"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="destinationType"> The Type being queried for support. </param>
+        public override bool CanConvertTo(ITypeDescriptorContext typeDescriptorContext, Type destinationType)
+        {
+            return destinationType == typeof(string);
+        }
+
+        /// <summary>
+        /// ConvertFrom - Attempt to convert to a RowDefinitionCollection from the given object.
+        /// </summary>
+        /// <returns>
+        /// The object which was constructed.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// An ArgumentNullException is thrown if the example object is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// An ArgumentException is thrown if the object is not null and is not a valid type,
+        /// or if the destinationType isn't one of the valid destination types.
+        /// </exception>
+        /// <param name="typeDescriptorContext"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="cultureInfo"> The CultureInfo which is respected when converting. </param>
+        /// <param name="value"> The Thickness to convert. </param>
+        public override object ConvertFrom(ITypeDescriptorContext typeDescriptorContext, CultureInfo cultureInfo, object value)
+        {
+            if (value != null && value is string input)
```

Comment:
This can be just `if (value is string input)`, since pattern matching discards anything that doesn't represent an actual value for the type (and `null` isn't considered a value), even in compile-time defined types. It makes the code clearer once you're used to it. Same in the other converter.

That is also why in case of nullable value types, you can just use pattern matching to avoid the `.Value` and `.HasValue` syntax in the underlying block.

Reply:
Fixed it